src/maxdiffusion/generate_wan.py

```diff
@@ -96,12 +123,20 @@ def inference_generate_video(config, pipeline, filename_prefix=""):
 
 def run(config, pipeline=None, filename_prefix=""):
   print("seed: ", config.seed)
-  from maxdiffusion.checkpointing.wan_checkpointer import WanCheckpointer
+  model_key = _MODEL_NAME.value
+
+  if model_key == "wan2.1":
+    from maxdiffusion.checkpointing.wan_checkpointer import WanCheckpointer
+  elif model_key == "wan2.2":
+    from maxdiffusion.checkpointing.wan_checkpointer2_2 import WanCheckpointer
 
   checkpoint_loader = WanCheckpointer(config, "WAN_CHECKPOINT")
-  pipeline = checkpoint_loader.load_checkpoint()
+  pipeline, opt_state, step = checkpoint_loader.load_checkpoint()
```

Comment:
this can probably be pipeline, _, _ = checkpoint_loader.load_checkpoint() as the variables are not used.